src/components/PostDetails.tsx

```diff
@@ -2,105 +1,132 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
+import { Post } from '../types/Post';
 import { NewCommentForm } from './NewCommentForm';
+import { getComments, addComment, deleteComment } from '../utils/fetchClient';
+import { Comment, CommentData } from '../types/Comment';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
-
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+type Props = {
+  currPost: Post | null;
+  openModal: boolean;
+  setOpenModal: (value: boolean) => void;
+};
 
-        <div className="block">
-          <Loader />
+export const PostDetails: React.FC<Props> = ({
+  currPost,
+  openModal,
+  setOpenModal,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [error, setError] = useState(false);
+  const [btnLoader, setBtnLoader] = useState(false);
+  const [commentLoader, setCommentLoader] = useState(false);
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+  const onAdd = ({ name, email, body }: CommentData) => {
+    setBtnLoader(true);
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+    const postId = currPost?.id || 0;
 
-          <p className="title is-4">Comments:</p>
+    return addComment({ postId, name, email, body })
+      .then(newComment => setComments([...comments, newComment]))
+      .catch(() => {
+        setError(true);
+      })
```

Comment:
The error handling in the `onAdd` function is quite generic. Consider providing more informative feedback or logging for debugging purposes.